sklearn/metrics/_ranking.py

```diff
@@ -239,6 +239,14 @@ def _binary_uninterpolated_average_precision(
     )
 
 
+@validate_params(
+    {
+        "y_true": ["array-like"],
+        "y_score": ["array-like"],
+        "pos_label": ["array-like", None],
+        "sample_weight": [Interval(Integral, 1, None, closed="left"), None],
```

Comment:
you switched the constraints of these 2 parameters :)
In addition, pos_label can be any integer or string so it should be ``[Integral, str, None]``